hi-ml-histopathology/testhisto/testhisto/utils/test_plots_utils.py

```diff
@@ -0,0 +1,86 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+from typing import Set
+from unittest.mock import MagicMock, patch
+import pytest
+from histopathology.utils.naming import ModelKey, PlotOptionsKey
+from histopathology.utils.plots_utils import DeepMILPlotsHandler
+from histopathology.utils.tiles_selection_utils import SlideNode, TilesSelector
+
+
+def test_plots_handler_wrong_plot_options() -> None:
+    plot_options = {PlotOptionsKey.HISTOGRAM, "foo"}
+    with pytest.raises(ValueError) as ex:
+        _ = DeepMILPlotsHandler(plot_options)  # type: ignore
+    assert "The selected plot option is not a valid option" in str(ex)
+
+
+def test_plots_handler_wrong_class_names() -> None:
+    plot_options = {PlotOptionsKey.HISTOGRAM, PlotOptionsKey.CONFUSION_MATRIX}
+    with pytest.raises(ValueError) as ex:
+        _ = DeepMILPlotsHandler(plot_options, class_names=[])
+    assert "No class_names were provided while activating confusion matrix plotting." in str(ex)
+
+
+def assert_plot_func_called_if_among_plot_options(
+    mock_plot_func: MagicMock, plot_option: PlotOptionsKey, plot_options: Set[PlotOptionsKey]
+) -> int:
+    calls_count = 0
+    if plot_option in plot_options:
+        mock_plot_func.assert_called()
+        calls_count += 1
+    else:
+        mock_plot_func.assert_not_called()
+    return calls_count
+
+
+@pytest.mark.parametrize(
+    "plot_options",
+    [
+        {},
+        {PlotOptionsKey.HISTOGRAM},
+        {PlotOptionsKey.HISTOGRAM, PlotOptionsKey.CONFUSION_MATRIX},
+        {PlotOptionsKey.HISTOGRAM, PlotOptionsKey.TOP_BOTTOM_TILES, PlotOptionsKey.SLIDE_THUMBNAIL_HEATMAP},
+        {
+            PlotOptionsKey.HISTOGRAM,
+            PlotOptionsKey.CONFUSION_MATRIX,
+            PlotOptionsKey.TOP_BOTTOM_TILES,
+            PlotOptionsKey.SLIDE_THUMBNAIL_HEATMAP,
+        },
+    ],
+)
+@patch("histopathology.utils.plots_utils.save_confusion_matrix")
+@patch("histopathology.utils.plots_utils.save_scores_histogram")
+@patch("histopathology.utils.plots_utils.save_top_and_bottom_tiles")
+@patch("histopathology.utils.plots_utils.save_slide_thumbnail_and_heatmap")
+def test_plots_handler_plots_only_desired_plot_options(
+    mock_slide: MagicMock,
+    mock_tile: MagicMock,
+    mock_histogram: MagicMock,
+    mock_conf: MagicMock,
```

Comment:
[Minor] Relying on matching the (reversed) orders of arguments and decorators looks a bit fragile. Might there be a safer way to do this?

Reply:
I personally find mocks written as context managers much easier to read.

Reply:
Changed with context managers.